core/src/main/java/org/infinispan/remoting/transport/Transport.java

```diff
@@ -146,6 +146,15 @@
     */
    int getViewId();
 
+   /**
+    * @return A {@link CompletableFuture} that completes when the transport has installed the expected view.
+    */
+   CompletableFuture<Void> viewFuture(int expectedViewId);
```

Comment:
I don't think so. While `viewFuture` might not be the best one, `view()` seems like returning a list of members (or similar structure), while this is `CF<Void>`. If you want to avoid "future", it could be something like `whenView(...)` or `withView(...)` which can be used in fluent syntax: `whenView(...).thenApply(...)` (ok, that's not really Shakespeare's style...)